bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanBridgeHandler.java

```diff
@@ -364,12 +394,76 @@ public void removePacketListener(PacketListener listener, long senderIdToListenT
         }
     }
 
-    public void startDiscovery(PacketListener teachInListener) {
+    public void startDiscovery(TeachInListener teachInListener) {
         transceiver.startDiscovery(teachInListener);
+
+        if (smackAvailable) {
+            // activate smack teach in
+            logger.debug("activate smack teach in");
+            try {
+                transceiver.sendBasePacket(ESP3PacketFactory.SA_WR_LEARNMODE(true),
+                        new ResponseListenerIgnoringTimeouts<BaseResponse>() {
+
+                            @Override
+                            public void responseReceived(BaseResponse response) {
+
+                                if (response.isOK()) {
+                                    logger.debug("Smack teach in activated");
+                                }
+                            }
+                        });
+            } catch (IOException e) {
+                logger.debug("Exception during activating smack teach in: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Smack packet could not be send");
```

Comment:
The log message could be removed, as the status update is already logged by the framework. You could include the exception's message in the status detail message.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Same for below.

Reply:
I am not really sure what you mean with status detail message. I hope my change is ok now. I changed this and other occurences. Thanks for the hint 👍